src/servers/multi_server.cc

```diff
@@ -0,0 +1,973 @@
+    }
+  }
+#ifndef TRITON_ENABLE_GPU
+  if (enforce_memory_type && requested_memory_type != TRITONSERVER_MEMORY_CPU) {
+    Usage(argv, "-m can only be set to \"system\" without enabling GPU");
+  }
+#endif  // TRITON_ENABLE_GPU
+
+  // Check API version.
+  uint32_t api_version_major, api_version_minor;
+  FAIL_IF_ERR(
+      TRITONSERVER_ApiVersion(&api_version_major, &api_version_minor),
+      "getting Triton API version");
+  if ((TRITONSERVER_API_VERSION_MAJOR != api_version_major) ||
+      (TRITONSERVER_API_VERSION_MINOR > api_version_minor)) {
+    FAIL("triton server API version mismatch");
+  }
+
+  // Create two instances of the server with 1 common and 1 unique repo each
+  TRITONSERVER_ServerOptions *server1_options = nullptr,
+                             *server2_options = nullptr;
+  SetServerOptions(
+      &server1_options, verbose_level, model_repository_paths[0],
+      model_repository_paths[1]);
+  SetServerOptions(
+      &server2_options, verbose_level, model_repository_paths[0],
+      model_repository_paths[2]);
+
+  TRITONSERVER_Server *server1_ptr = nullptr, *server2_ptr = nullptr;
+  FAIL_IF_ERR(
+      TRITONSERVER_ServerNew(&server1_ptr, server1_options),
+      "creating first server instance");
+  FAIL_IF_ERR(
+      TRITONSERVER_ServerNew(&server2_ptr, server2_options),
+      "creating second server instance");
+  FAIL_IF_ERR(
+      TRITONSERVER_ServerOptionsDelete(server1_options),
+      "deleting server options");
+  FAIL_IF_ERR(
+      TRITONSERVER_ServerOptionsDelete(server2_options),
+      "deleting server options");
+
+  std::shared_ptr<TRITONSERVER_Server> server1(
+      server1_ptr, TRITONSERVER_ServerDelete);
+  std::shared_ptr<TRITONSERVER_Server> server2(
+      server2_ptr, TRITONSERVER_ServerDelete);
+
+  // Wait and until the servers are both live and ready.
+  CheckServerLiveAndReady(server1);
+  CheckServerLiveAndReady(server2);
+
+  // Print status of the servers.
+  PrintServerStatus(server1);
+  PrintServerStatus(server2);
+
+  // Load models in both servers.
+  FAIL_IF_ERR(
+      TRITONSERVER_ServerLoadModel(server1.get(), "simple1"),
+      "failed to load model");
+  FAIL_IF_ERR(
+      TRITONSERVER_ServerLoadModel(server1.get(), "simple2"),
+      "failed to load model");
+  FAIL_IF_ERR(
+      TRITONSERVER_ServerLoadModel(server2.get(), "simple1"),
+      "failed to load model");
+  FAIL_IF_ERR(
+      TRITONSERVER_ServerLoadModel(server2.get(), "simple3"),
+      "failed to load model");
+
+  // Wait for the models to become available.
+  bool is_int1 = true, is_int2 = true, is_int3 = true;
+  bool is_torch_model1 = false, is_torch_model2 = false,
+       is_torch_model3 = false;
+  AwaitModelReady(server1, "simple1", &is_int1, &is_torch_model1);
+  AwaitModelReady(server1, "simple2", &is_int2, &is_torch_model2);
+  AwaitModelReady(server2, "simple1", &is_int1, &is_torch_model1);
+  AwaitModelReady(server2, "simple3", &is_int3, &is_torch_model3);
```

Comment:
As an example, it can be more specific so that it is simpler.. We know that the models in repo is in float so we can remove `is_int`, and I would say use other framework instead of libtorch to avoid additional handling on IO names

Reply:
As discussed with @deadeyegoodwin we thought it would be good to cover multiple backends (except TRT)

Reply:
Why except TRT? Besides that, we do have at least 3 different backends even excluding TRT and Torch, right? If not, then we have to keep libtorch in the example, it is just annoying to follow the additional naming convention.

Reply:
Replacing libtorch with plan for now. I agree removing libtorch makes it better.